### PR TITLE
Feature/a43 Add edit user password to member settings

### PR DESCRIPTION
**Pull request by cm8263**
This PR allows admins to change a user's password to a temp password of their choosing. Temp passwords must be at least 3 characters long.

This PR adds a `editpassword` method to the  Edit Profile controller; adds custom css to the custom.css file to override css injected by js by sweetalerts; adds a sweetalert to the js file for the edit profile model; adds a new route for the previously mentioned controller.

### CODE REVIEW

File: public/js/modals/edit_profile_modal_EDIT.js [hunk 1]
```diff
@@ -68,4 +68,68 @@ $('#ajax_edit_member').on('submit', function(e) {
...
+          }
+        }
+    }).then(newPassword => {
+        let modalNode = $('.js-swal-fixed')
+        if (!modalNode) return;
+
+        modalNode.attr('tabindex', '-1');
+        modalNode.removeClass('js-swal-fixed');
+
+        if(newPassword.length < 4) {
```
Comment (Redbully14) [inline public/js/modals/edit_profile_modal_EDIT.js]: Make the check for at least 8 characters.

File: app/Http/Controllers/Auth/EditProfileController.php [hunk 2]
```diff
@@ -113,4 +113,15 @@ public function edit(Request $request)
...
         return;
     }
-}
\ No newline at end of file
+
+    public function editpassword(Request $request)
+    {
+        $user = User::find($request->route('user'));
+
+        $user->password = Hash::make($request->new_password);
```
Comment (Redbully14) [inline app/Http/Controllers/Auth/EditProfileController.php]: Make sure to pass the request through a backend validator as well, just check that it's required, a string and at least 8 characters long.

File: resources/views/modals/edit_profile_modal.blade.php [hunk 3]
```diff
@@ -219,8 +231,10 @@
 @if(Auth::user()->level() >= $constants['access_level']['seniorstaff'])
 <script type="text/javascript">
   var $url_edit_profile_modal_POST = '{{ url('member/edit/edit_user/') }}/';
+  var $url_edit_profile_password_modal_POST = '{{ url('member/edit/edit_user_password/') }}/';
 </script>
+<script src="/assets/vendors/sweetalert/sweetalert.min.js"></script>
```
Comment (Redbully14) [inline resources/views/modals/edit_profile_modal.blade.php]: I'd move this over to master/app.blade.php

I see it being used in the future for other stuff as well.

File: resources/views/modals/edit_profile_modal.blade.php [hunk 4]
```diff
@@ -188,6 +188,18 @@
                     </div>
                     @endif
 
+                    @if(Auth::user()->level() >= $constants['access_level']['admin'])
+                    <div class="form-group">
+                      <label>Antelope Password</label>
+                      <button type="button" class="form-control btn btn-outline-warning btn-fw" onclick="changePasswordPopup()">Change User Password</button>
+                    </div>
+                    @else
```
Comment (Redbully14) [inline resources/views/modals/edit_profile_modal.blade.php]: Remove everything below the else handle (remove line 196-200)
There really isn't a point for there to be an Antelope username field hidden below the configuration above.
If you inspect element, there's two of them hidden twice now:
image

Comment (cm8263) [reply resources/views/modals/edit_profile_modal.blade.php]: Yeah I didn't mean to leave that in, good spot.

**Pull request merged**